stake-pool/js/src/test.ts

```diff
@@ -0,0 +1,194 @@
+import * as index from './index.js';
+import * as schema from './schema.js';
+import BN from 'bn.js';
+import assert from 'assert';
+import {PublicKey, Connection} from '@solana/web3.js';
+
+describe('schema.decode', () => {
+  describe('StakePoolAccount', () => {
+    it('should successfully decode StakePoolAccount account data', () => {
+      const decodedData = schema.StakePool.decode(
+        new Buffer(
+          '014f962536a1b5b3a1f1036599ca7c164d6ad0677fed896f12ecdd09ada2cd23dc4f962536a1b5b3a1f1036599ca7c164d6ad0677fed896f12ecdd09ada2cd23dcbb09705cd25f4ad3b9e2c78e3f62ba3e290f798150499edc4f851a1a45e71159ff1086831390e666d64d18cac8f0c86eb91d799eda647319634280b89548ba0371ee618745ef3ebb93164e66ac3560e934b04e49b89ad38baa3b104284abb1a574003e566772f55676cf243c647fab463afa138cb156d2080062d2957b10bf22870fdbe150fc449db6268eb9214a848a90162d9bd4bb81e722cacb41d3dcb483b706ddf6e1d765a193d9cbe146ceeb79ac1cb485ed5f5b37913a8cf5857eff00a9000000000000000000000000000000007c00000000000000e8030000000000003800000000000000',
```

Comment:
this test may be slightly brittle... I've since added some fields, which would break these tests, and modifying them is pretty annoying.  Maybe you can test the round-trip logic of creating the type, serializing, and deserializing?

Reply:
Not sure what you mean here: wouldn't adding fields break the entire deserialiser, not just the tests?

Reply:
solved in https://github.com/solana-labs/solana-program-library/pull/1656/commits/e122e32c348ef0ec449b11e0715e32a17092e9bc